lib/mp4/transmuxer.js

```diff
@@ -663,13 +740,23 @@ Transmuxer = function(options) {
 
   // feed incoming data to the front of the parsing pipeline
   this.push = function(data) {
-    packetStream.push(data);
+    if (aacfile === undefined) {
```

Comment:
You can get rid of this whole condition if you just saved whatever object was the head of the pipeline to a single variable. In other words, on push you should always push data into `pipelineRoot` or some such. Which object it actually is will be determined by which setup function was called.